package_metrics/package_managers/utils.py

```diff
@@ -0,0 +1,27 @@
+from package_metrics.constants import PIP
+from package_metrics.package_managers.pip import get_pip_packages
+from package_metrics.parsing_utils import behind
+
+
+def iter_packages(package_manager):
+    """
+    Obtains packages given a manager, and builds package info tuple
+    :param package_manager: str representing package manager
+    :return: generator of package info (behind, name, latest, current)
+    """
+    for package_info in get_packages(package_manager):
+        latest = package_info["latest_version"]
+        current = package_info["version"]
+        yield behind(latest, current), package_info["name"], latest, current
+
+
+def get_packages(package_manager):
+    """
+    :param package_manager: str representing package manager
+    :return: function to retrieve packages for the provided package manager
+    """
+    package_list_map = {
+        PIP: get_pip_packages,
```

Comment:
Is there someplace where we can document what should `get_<package_manager>_packages` should return?
The format of individual package objects?

This can get handy if we try to add a new package manager then we would know what our transformation function should spit out.

Reply:
Yeah that is a good idea. I could create a README for the `package_managers` module that details adding support for a new package manager.

Reply:
See 2aa10554aa9c1300f53f3454398bfdfc46965ab1